src/main/scala/intellij/haskell/action/DownloadCabalConfigFile.scala

```diff
@@ -0,0 +1,17 @@
+package intellij.haskell.action
+
+import com.intellij.openapi.actionSystem.{AnAction, AnActionEvent}
+import intellij.haskell.external.component.HaskellComponentsManager.refreshGlobalProjectInfo
+import intellij.haskell.external.component.PathComponent
+
```

Comment:
I think there is no need to download the cabal.config file every time a project is opened because we can get the old LTS release from the local cabal.config file (if it exists), so only when the cabal.config file is not downloaded or the resolver is changed, we have to download the file again. Downloading file is really time-consuming, so I believe this approach is reasonable enough. And another reason why this action exists here is that if users can not download the cabal.config file and load all the package names into global project info, they can just use this action directly without reopening the project, 
>When package completion is requested and there is no cabal.config file it has to be downloaded. Because of the case user changes Stackage LTS release, every time project is opened the cabal.config has to be downloaded. Downloading library sources is only done at request and when creating project because it takes some time.